### PR TITLE
Add appversion filtering to the search API and display compatibility info (fix #2552)

### LINKED ISSUE #2552: Filter search API by appversion
After #2545 we'll also need to filter by `appversion`. This is lower priority because it's not needed for internal search but this would be nice to have to be able to replicate the website search results.

### PR DESCRIPTION
**Pull request by diox**
#2552

### CODE REVIEW

File: src/olympia/addons/tests/test_indexers.py [hunk 1]
```diff
@@ -120,8 +120,10 @@ def test_extract_attributes(self):
         assert extracted['app'] == [FIREFOX.id]
         assert extracted['appversion'] == {
             FIREFOX.id: {
-                'min': self.addon.compatible_apps[FIREFOX].min.version_int,
-                'max': self.addon.compatible_apps[FIREFOX].max.version_int,
+                'min': 2000000200100L,
+                'max': 4000000200100L,
+                'max_human': '4.0',
+                'min_human': '2.0',
```
Comment (diox) [inline src/olympia/addons/tests/test_indexers.py]: See, mapping tests do work :) This was failing before I adjusted it.

File: src/olympia/addons/indexers.py [hunk 2]
```diff
@@ -23,6 +23,8 @@ def get_mapping(cls):
             'properties': {
                 'max': {'type': 'long'},
                 'min': {'type': 'long'},
+                'max_human': {'type': 'string', 'index': 'no'},
+                'min_human': {'type': 'string', 'index': 'no'},
```
Comment (diox) [inline src/olympia/addons/indexers.py]: I could have done without that and converted back the version integer to a string in the serializer, but it turns out that the code we have to convert a version integer back to a string is not 100% fool-proof, it has issues with trailing zeros and `*`, so I preferred just storing the actual human-readable version.

Comment (EnTeQuAk) [inline src/olympia/addons/indexers.py]: that sounds good to me, and there's still `max` and `min` in the index if we have the need for it.

File: src/olympia/addons/tests/test_indexers.py (hunk 1 above)
Comment (EnTeQuAk) [inline src/olympia/addons/tests/test_indexers.py]: 😆

**Comment by EnTeQuAk** [comment]
r+wc

File: src/olympia/addons/indexers.py (hunk 2 above)
Comment (diox) [inline src/olympia/addons/indexers.py]: It's only meant to be displayed back to the client in the API in the `compatibility` field, so it does not care what the version string is, it'll just store it and return it as-is.

**Pull request merged**